controllers/deactivation/deactivation_controller.go

```diff
@@ -128,6 +146,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		if err := r.resetDeactivatingState(ctx, usersignup); err != nil {
 			return reconcile.Result{}, err
 		}
+		// Set the scheduled deactivation time to nil
+		if usersignup.Status.ScheduledDeactivationTimestamp != nil {
```

Comment:
You don't need to check if it's already nil or not. The status updater checks the old and new value.

Reply:
Good catch, fixed